lib/optimizer/src/Transformer/ServerSideRendering.php

```diff
@@ -620,4 +699,260 @@ private function isAmpExperimentUsed(DOMElement $element)
+     * @param string   $customCss Custom CSS styling to add.
+     * @return bool Whether adding the custom CSS styling to the document succeeded.
+     */
+    private function addCustomCss(Document $document, $customCss)
+    {
+        $additionalBytes = strlen($customCss);
+
+        if (($this->ampCustomCssByteCount + $additionalBytes) > self::MAX_CSS_BYTE_COUNT) {
+            return false;
+        }
+
+        if (empty($this->ampCustomStyleElement)) {
+            $ampCustomStyleElement = $document->xpath->query(self::STYLE_AMP_CUSTOM_XPATH, $document->head)->item(0);
+            if ($ampCustomStyleElement instanceof DOMElement) {
+                $this->ampCustomStyleElement = $ampCustomStyleElement;
+                $this->ampCustomCssByteCount = (new CssByteCountCalculator($document))->calculate();
+                if (($this->ampCustomCssByteCount + $additionalBytes) > self::MAX_CSS_BYTE_COUNT) {
+                    return false;
+                }
+            } else {
+                $ampCustomStyleElement = $document->createElement(Tag::STYLE);
+                $ampCustomStyleElement->setAttribute(Attribute::AMP_CUSTOM, null);
+                $this->ampCustomStyleElement = $ampCustomStyleElement;
+                $document->head->appendChild($this->ampCustomStyleElement);
+            }
+        }
+
+        $this->ampCustomStyleElement->textContent .= $customCss;
+        $this->ampCustomCssByteCount              += strlen($customCss);
+
+        return true;
+    }
+
+    /**
+     * Extract the custom CSS styling from a 'sizes' attribute.
+     *
+     * __sizes__
+     * "One or more strings separated by commas, indicating a set of source sizes. Each source size consists of:
+     * - A media condition. This must be omitted for the last item in the list.
+     * - A source size value."
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-sizes
+     *
+     * @param Document   $document       Document containing the element to adapt.
+     * @param DOMElement $element        Element to adapt.
+     * @param string     $attributeValue Value of the attribute to extract the CSS styling from.
+     * @return string Extract custom CSS styling.
+     */
+    private function extractSizesAttributeCss(Document $document, DOMElement $element, $attributeValue)
+    {
+        if (!$element->hasAttribute(Attribute::SRCSET) || empty($element->getAttribute(Attribute::SRCSET))) {
+            // According to the Mozilla docs, a sizes attribute without a valid srcset attribute should have no effect.
+            // Therefore, it should simply be stripped, without producing media queries.
+            // @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-sizes
+            return '';
+        }
+
+        return $this->extractAttributeCss(
+            $document,
+            $element,
+            $attributeValue,
+            '#__ID__{width:%s};',
```

Comment:
Semicolon should be removed here, no?

```suggestion
            '#__ID__{width:%s}',
```